### PR TITLE
Updated theme & extended documentation

### PR DESCRIPTION
**Pull request by MyIgel**
* Updated the theme to the last working version
* Re-ordered the menu items
* Updated some texts
* Added section for API documentation
* Added preliminary development documentation & request lifecycle page

### CODE REVIEW

File: content/admin/configuration.md [hunk 1]
```diff
@@ -111,7 +113,7 @@ Look at the default as sample:
 
 ```php
     // Available themes
```
Comment (xuwhite) [inline content/admin/configuration.md]: some available themes pls

File: content/admin/configuration.md [hunk 2]
```diff
@@ -177,11 +169,14 @@ Look at the default as sample:
...
 
-## Enable t-shirts
+## Enable goodies/t-shirts
 
 ```php
-    // Enables the T-Shirt configuration on signup and profile
-    'enable_tshirt_size'      => true,
+    // Resembles the Goodie Type. There are three options:
+    // 'none' => no goodie at all
+    // 'goodie' => a goodie which has no sizing options
```
Comment (xuwhite) [inline content/admin/configuration.md:178]: fix pls

File: content/admin/configuration.md [hunk 3]
```diff
@@ -177,11 +169,14 @@ Look at the default as sample:
     'min_password_length'     => 8,
 ```
 
-## Enable t-shirts
+## Enable goodies/t-shirts
 
 ```php
-    // Enables the T-Shirt configuration on signup and profile
-    'enable_tshirt_size'      => true,
+    // Resembles the Goodie Type. There are three options:
```
Comment (xuwhite) [inline content/admin/configuration.md:176]: fix goody pls

File: content/admin/configuration.md [hunk 4]
```diff
@@ -177,11 +169,14 @@ Look at the default as sample:
...
+    // 'none' => no goodie at all
+    // 'goodie' => a goodie which has no sizing options
+    // 'tshirt' => goodie that is called tshirt and has sizing options
+    'goodie_type'      => 'tshirt',
```
Comment (xuwhite) [inline content/admin/configuration.md:180]: fix goody pls

File: content/api/_index.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+---
+title: "API"
+date: 2023-02-04T01:20:00+01:00
+weight: 70
+---
+
+{{% notice warning %}}
+The Engelsystem contains a work in progress **beta** API.
+It might change at any time without further notice but should be "stable" during an event.
```
Comment (xuwhite) [inline content/api/_index.md]: what event? when?

File: content/api/_index.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+{{% /notice %}}
+
+The API specification can be found [here]({{% ref "api/specification" %}}).
+
+The API is accessible to Bureaucrats or users in the `API` group to allow for testing but to not confuse regular users.
```
Comment (xuwhite) [inline content/api/_index.md]:
```suggestion
The API is accessible to users in the `API` or ` Bureaucrat` group to allow for testing but to not confuse regular users.
```

File: content/developer/_index.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+---
+title: "Development"
+date: 2024-04-08T22:10:00+02:00
+weight: 60
+---
+
+This documentation provides some additional background on extending and hacking around the Engelsystem.
```
Comment (xuwhite) [inline content/developer/_index.md:7]: does not work for me

File: content/developer/_index.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+and [contributing](https://github.com/engelsystem/engelsystem/blob/main/CONTRIBUTING.md) guidelines
+for a basic setup and guidance.
+
+## Design Philosophy
+
+The Engelsystem functionality and code structure after the refactoring is loosely inspired by Laravel
+(and uses some Laravel components).
+We try to provide a somehow stable version in our `main` branch that "Just Works" and has the newest features.
+
+New Versions are normally released after bigger events like the Chaos Communication Congress and thus
```
Comment (xuwhite) [inline content/developer/_index.md]: events?

File: content/developer/_index.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+at least once a year.
+External dependencies should do their function well and be actively maintained.
```
Comment (xuwhite) [inline content/developer/_index.md]:
```suggestion
External dependencies should fulfill their function well and be actively maintained.
```

File: content/developer/request-lifecycle.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
+---
+title: "Request Lifecycle"
+date: 2024-04-08T22:20:00+02:00
+---
+
+Our goal with this overview is to give you a basic understanding how the Engelsystem handles a request
```
Comment (xuwhite) [inline content/developer/request-lifecycle.md]:
```suggestion
The goal of this overview is to give you a basic understanding of how the Engelsystem handles a request
and provide a way to troubleshoot problems if they occur.
```

**Review by xuwhite — CHANGES_REQUESTED** [review]
So Long, and Thanks for All the Fish

File: content/developer/_index.md (hunk 7 above)
Comment (MyIgel) [reply content/developer/_index.md:7]: Then feel free to improve it ;)

**Review by xuwhite — APPROVED** [review]
looks good to me

**Pull request merged**